baybe/surrogates/base.py

```diff
@@ -226,24 +226,52 @@ def fit(
         # Create scaler objects
         input_scaler = self._make_input_scaler(searchspace)
         output_scaler = self._make_output_scaler(objective, measurements)
-        self._output_scaler = output_scaler
 
         def transform_inputs(df: pd.DataFrame, /) -> pd.DataFrame:
             """Fitted input transformation pipeline."""
+            # IMPROVE: This method currently relies on two workarounds required
+            #   due the working mechanism of sklearn's ColumnTransformer:
+            #   * Unfortunately, the transformer returns a raw array, meaning that
+            #       column names need to be manually attached afterward.
+            #   * In certain cases (e.g. in hybrid spaces), the method needs
+            #       to transform only a subset of columns. Unfortunately, it is not
+            #       possible to use a subset of columns once the transformer is set up,
+            #       which is a side-effect of the first point. As a workaround,
+            #       we thus fill the missing columns with NaN and subselect afterward.
+
+            # For the workaround, collect all comp rep columns of the parameters
+            # that are actually present in the given dataframe. At the end,
+            # we'll filter the transformed augmented dataframe down to these columns.
+            exp_rep_cols = [p.name for p in searchspace.parameters]
+            comp_rep_cols = []
+            for col in [c for c in df.columns if c in exp_rep_cols]:
```

Comment:
Why do we need the check for `exp_rep_columns`? Because `df` can be in computational representation?

Reply:
No, because `df` can also contain arbitrary other columns, and indeed contains target columns in some context. So without the restriction, it would just break.